src/torrent.cpp

```diff
@@ -1530,141 +1527,27 @@ bool is_downloading_state(int const st)
-		auto* gens = static_cast<GENERAL_NAMES*>(
-			X509_get_ext_d2i(cert, NID_subject_alt_name, nullptr, nullptr));
-
-#ifndef TORRENT_DISABLE_LOGGING
-		std::string names;
-		bool match = false;
-#endif
-		for (int i = 0; i < aux::openssl_num_general_names(gens); ++i)
-		{
-			GENERAL_NAME* gen = aux::openssl_general_name_value(gens, i);
-			if (gen->type != GEN_DNS) continue;
-			ASN1_IA5STRING* domain = gen->d.dNSName;
-			if (domain->type != V_ASN1_IA5STRING || !domain->data || !domain->length) continue;
-			auto const* torrent_name = reinterpret_cast<char const*>(domain->data);
-			std::size_t const name_length = aux::numeric_cast<std::size_t>(domain->length);
-
-#ifndef TORRENT_DISABLE_LOGGING
-			if (i > 1) names += " | n: ";
-			names.append(torrent_name, name_length);
-#endif
-			if (std::strncmp(torrent_name, "*", name_length) == 0
-				|| std::strncmp(torrent_name, m_torrent_file->name().c_str(), name_length) == 0)
-			{
-#ifndef TORRENT_DISABLE_LOGGING
-				match = true;
-				// if we're logging, keep looping over all names,
-				// for completeness of the log
-				continue;
-#else
-				return true;
-#endif
-			}
-		}
-
-		// no match in the alternate names, so try the common names. We should only
-		// use the "most specific" common name, which is the last one in the list.
-		X509_NAME* name = X509_get_subject_name(cert);
-		int i = -1;
-		ASN1_STRING* common_name = nullptr;
-		while ((i = X509_NAME_get_index_by_NID(name, NID_commonName, i)) >= 0)
-		{
-			X509_NAME_ENTRY* name_entry = X509_NAME_get_entry(name, i);
-			common_name = X509_NAME_ENTRY_get_data(name_entry);
-		}
-		if (common_name && common_name->data && common_name->length)
-		{
-			auto const* torrent_name = reinterpret_cast<char const*>(common_name->data);
-			std::size_t const name_length = aux::numeric_cast<std::size_t>(common_name->length);
-
-#ifndef TORRENT_DISABLE_LOGGING
-			if (!names.empty()) names += " | n: ";
-			names.append(torrent_name, name_length);
-#endif
-
-			if (std::strncmp(torrent_name, "*", name_length) == 0
-				|| std::strncmp(torrent_name, m_torrent_file->name().c_str(), name_length) == 0)
-			{
-#ifdef TORRENT_DISABLE_LOGGING
-				return true;
-#else
-				match = true;
-#endif
-			}
-		}
-
-#ifndef TORRENT_DISABLE_LOGGING
-		debug_log("<== incoming SSL CONNECTION [ n: %s | match: %s ]"
-			, names.c_str(), match?"yes":"no");
-		return match;
-#else
-		return false;
-#endif
-	}
-
+#ifdef TORRENT_USE_SSL
 	void torrent::init_ssl(string_view cert)
 	{
-		using boost::asio::ssl::context;
-
-		// this is needed for openssl < 1.0 to decrypt keys created by openssl 1.0+
-#if !defined(OPENSSL_API_COMPAT) || (OPENSSL_API_COMPAT < 0x10100000L)
-		OpenSSL_add_all_algorithms();
-#else
-		OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_CIPHERS | OPENSSL_INIT_ADD_ALL_DIGESTS, nullptr);
-#endif
-
 		// create the SSL context for this torrent. We need to
 		// inject the root certificate, and no other, to
 		// verify other peers against
 #if BOOST_VERSION >= 106400
```

Comment:
in the latest `master`, boost older than 1.64 is no longer supported, so this was simplified a bit